source/coll_db.f90

```diff
@@ -632,6 +637,72 @@ subroutine cdb_writeDB_newFromOld
 
 end subroutine cdb_writeDB_newFromOld
 
+! ================================================================================================ !
+!  M. D'Andrea, BE-ABP-HSS
+!  Created: 2020-02-19
+!  Updated: 2020-02-20
+!  Determine most frequent sigma setting for each collimator family and use that as family setting
+!  Identify empty collimator families and set their aperture to zero for cross-checks
+! ================================================================================================ !
+
+subroutine cdb_reconstructNSigFam
+
+  use mod_alloc
+
+  integer j, k, count_sig, freq, freq_max
+  real(kind=fPrec), allocatable :: NSig_collFam(:)
+
+  do j=1,cdb_nFam
+
+    count_sig = 0
+    do k=1,cdb_nColl
+      if(cdb_cFamily(k) == j) then
+        count_sig = count_sig+1
+        call alloc(NSig_collFam,count_sig,cdb_cNSig(k),"NSig_collFam")
+      end if
+    end do
+
+    freq_max = 0
+    do k=1,count_sig
+      freq = count(NSig_collFam == NSig_collFam(k))
+      if(freq > freq_max) then
+        freq_max = freq
+        cdb_famNSig(j) = NSig_collFam(k)
+      end if
+    end do
+
+    if(count_sig /= 0) then
+      call dealloc(NSig_collFam,"NSig_collFam")
+    else
+      cdb_famNSig(j) = 0
```

Comment:
probably we can skip this, as the `cdb_nullEmptyFam` subroutine is called afterwards